pootle/apps/pootle_data/utils.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from django.contrib.auth import get_user_model
+from django.db import models
+from django.utils.functional import cached_property
+
+from pootle_statistics.models import SubmissionTypes
+from pootle_store.constants import FUZZY, OBSOLETE, TRANSLATED, UNTRANSLATED
+from pootle_store.models import Suggestion
+from pootle_store.util import SuggestionStates
+
+
+User = get_user_model()
+
+
+class StoreDataTool(object):
+    """Sets the stats for a store"""
+
+    def __init__(self, store):
+        self.store = store
+
+    @property
+    def units(self):
+        return self.store.unit_set.live()
+
+    @cached_property
+    def last_created_unit(self):
+        """Last unit created in this Store"""
+        return (
+            self.units.exclude(creation_time__isnull=True)
+                      .order_by('-creation_time', '-revision').first())
```

Comment:
but we update in consequence, that's why I'm asking. probs revision is unnecessary if there can be the cases which are of no consequence.